client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -0,0 +1,144 @@
+import { DataForm } from '@automattic/dataviews';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState } from 'react';
+import type { SiteSettings } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+import type { UseMutationResult } from '@tanstack/react-query';
+
+interface SitePrivacy {
+	visibility: string;
+}
+
+const fields: Field< SitePrivacy >[] = [
+	{
+		id: 'visibility',
+		Edit: ( { field, onChange, data } ) => {
```

Comment:
What's special about this field that it requires a dedicated "Edit" component. Can't we just have a builtin control (a "type" and a Edit: "toggle" or something like that).

Reply:
I don't think DataForm supports `<ToggleGroupControl>`:

https://github.com/Automattic/wp-calypso/tree/d102f9f5e7a23bb265b0e501b2d51016a8b0acbc/packages/dataviews/src/dataform-controls

Reply:
Let's implement a toggle control there then.

Reply:
OK, finally I tried. But I can't match the design -- DataForm controls won't support label-less fields; it will be the `id` if not provided:

https://github.com/Automattic/wp-calypso/blob/636484c11ed52ecfde985cb2af1fcfc946f52474/packages/dataviews/src/normalize-fields.ts#L72

whereas, in our hosting dashboard, the site settings page will be individual settings, and thus we won't need the field label (in most cases):

<img width="627" alt="image" src="https://github.com/user-attachments/assets/0fc237af-cce2-4509-bf52-39b7a3309d0c" />

<img width="635" alt="image" src="https://github.com/user-attachments/assets/c8a76c8f-f168-4a49-a446-c780617a58c5" />

Reply:
DataForm supports `labelPosition` top|left maybe we can also support position "none" (so hidden visually). Did we consider that? I know I've proposed this a long time ago in Gutenberg, not sure where we landed there cc @louwie17 @oandregal.

There is always a possibility of saying this is not a DataForm since it's a single field but if I'm not wrong we don't want multiple fields here with conditions later.

Reply:
Oh... I managed to use `labelPosition: 'none'` in the form fields... let me try again.

Reply:
@youknowriad I believe `none` is supported already if you look at the Storybook example: https://wordpress.github.io/gutenberg/?path=/docs/dataviews-dataform--docs
https://github.com/WordPress/gutenberg/blob/c38610ae21d5aad266aff9a08fc49e80cd38f0e6/packages/dataviews/src/dataforms-layouts/regular/index.tsx#L111

Reply:
Should be able to do:
```
fields: [
				{
					id: 'title',
					labelPosition: 'none'
				}
]
```

Reply:
Raised: https://github.com/Automattic/wp-calypso/pull/103483

Reply:
Updated to use the new `toggleGroup` control.